syntax = "proto3";

package grpcNewsPubSub;

service MsgServUnarySub{
  // Obtains the MessageResponse at a given position.
  rpc GetServerResponse(SubscribeRequest) returns (NewsArticle) {}
}

service MsgServPubStreamSubUnary {
  rpc GetServerResponse(SubscribeRequest) returns (stream NewsArticle);
}

message SubscribeRequest {
    string subscriber_id = 1;
    repeated string topics = 2;
}

message NewsArticle {
    repeated string title = 1;
    repeated string description = 2;
    repeated string url = 3;
    repeated string published_at = 4;
}
